addHotelProcess.php

```diff
@@ -0,0 +1,67 @@
+<?php
+require 'connection.php';
+
+$jsonText = $_POST["hotelData"];
+$dataObject = json_decode($jsonText);
+
+$name = $dataObject->hotelName;
+$address = $dataObject->hotelAddress;
+$email = $dataObject->email;
+$contact = $dataObject->contact;
+$numOfRooms = $dataObject->numberOfRooms;
+$roomNumbers = explode(",", $dataObject->roomNumbers);
+$roomType = $dataObject->roomType;
+$roomPrice = $dataObject->pricePerRoom;
+
+if (empty($name)) {
+    echo "Enter the name of the hotel";
+} else if (strlen($name) > 45) {
+    echo "Hotel name is too long";
+} else if (empty($address)) {
+    echo "Enter the hotel address";
+} else if (strlen($address) > 100) {
+    echo "Hotel address is too long";
+} else if (empty($email)) {
+    echo "Enter the email";
+} else if (strlen($email) > 45) {
+    echo "Email is too long";
+} else if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+    echo "Enter a valid email";
+} else if (empty($contact)) {
+    echo "Enter the contact number";
+} else if (strlen($contact) != 10) {
+    echo "Contact number must have 10 characters";
+} else if (!preg_match("/07[0,1,2,4,5,6,7,8][0-9]/", $contact)) {
+    echo "Enter a valid contact number";
+} else if (empty($numOfRooms)) {
+    echo "Enter the number of rooms";
+} else if ($numOfRooms < 0 || $numOfRooms === 0 || $numOfRooms > 10) {
```

Comment:
This `if condition` is not optimized. Add a optimized condition.

Reply:
OK